protocol/app/process/transactions.go

```diff
@@ -42,77 +59,58 @@ func DecodeProcessProposalTxs(
 ) (*ProcessProposalTxs, error) {
 	// Check len.
 	numTxs := len(req.Txs)
-	if numTxs < MinTxsCount {
+	if numTxs < minTxsCount {
 		return nil, errorsmod.Wrapf(
 			ErrUnexpectedNumMsgs,
 			"Expected the proposal to contain at least %d txs, but got %d",
-			MinTxsCount,
+			minTxsCount,
 			numTxs,
 		)
 	}
 
-	txTypeToIdx, idxToTxType := GetAppInjectedMsgIdxMaps(numTxs)
-
 	// Operations.
-	orderIdx, ok := txTypeToIdx[ProposedOperationsTxType]
-	if !ok {
-		panic("must define ProposedOperationsTxType")
-	}
-	operationsTx, err := DecodeProposedOperationsTx(decoder, req.Txs[orderIdx])
+	operationsTx, err := DecodeProposedOperationsTx(decoder, req.Txs[proposedOperationsTxIndex])
 	if err != nil {
 		return nil, err
 	}
 
 	// Acknowledge bridges.
-	acknowledgeBridgesIdx, ok := txTypeToIdx[AcknowledgeBridgesTxType]
-	if !ok {
-		panic("must define AcknowledgeBridgesTxType")
-	}
 	acknowledgeBridgesTx, err := DecodeAcknowledgeBridgesTx(
 		ctx,
 		bridgeKeeper,
 		decoder,
-		req.Txs[acknowledgeBridgesIdx],
+		req.Txs[numTxs+acknowledgeBridgesTxLenOffset],
 	)
 	if err != nil {
 		return nil, err
 	}
 
 	// Funding samples.
-	addFundingIdx, ok := txTypeToIdx[AddPremiumVotesTxType]
-	if !ok {
-		panic("must define AddPremiumVotesTxType")
-	}
-	addPremiumVotesTx, err := DecodeAddPremiumVotesTx(decoder, req.Txs[addFundingIdx])
+	addPremiumVotesTx, err := DecodeAddPremiumVotesTx(decoder, req.Txs[numTxs+addPremiumVotesTxLenOffset])
 	if err != nil {
 		return nil, err
 	}
 
 	// Price updates.
-	updatePricesIdx, ok := txTypeToIdx[UpdateMarketPricesTxType]
-	if !ok {
-		panic("must define AddPremiumVotesTxType")
-	}
-	updatePricesTx, err := DecodeUpdateMarketPricesTx(ctx, pricesKeeper, decoder, req.Txs[updatePricesIdx])
+	updatePricesTx, err := DecodeUpdateMarketPricesTx(
+		ctx,
+		pricesKeeper,
+		decoder,
+		req.Txs[numTxs+updateMarketPricesTxLenOffset],
+	)
 	if err != nil {
 		return nil, err
 	}
 
 	// Other txs.
-	allOtherTxs := make([]*OtherMsgsTx, numTxs-len(txTypeToIdx))
-	idx := 0
-	for i, txBytes := range req.Txs {
-		if _, exists := idxToTxType[i]; exists { // skip, because tx is not part of "others".
-			continue
-		}
-
+	allOtherTxs := make([]*OtherMsgsTx, numTxs-minTxsCount)
+	for i, txBytes := range req.Txs[firstOtherTxOffset : numTxs+lastOtherTxLenOffset] {
```

Comment:
do we want some sort of sanity-check that `numTxs-minTxsCount == (numTxs+lastOtherTxLenOffset) - firstOtherTxOffset` ?

Is it preferable to continue in this case?

Reply:
added `minTxsCount != firstOtherTxOffset-lastOtherTxLenOffset` to init for our sanity check.